### PR TITLE
libnghttp2 nghttp2 1.62.1

### PR DESCRIPTION
**Pull request by bevanjkay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by chenrui333 — APPROVED** [review]

**Review by iMichka — APPROVED** [review]

**Comment by bevanjkay** [comment]
Upstream issue for GCC11 build failure on Linux - https://github.com/nghttp2/nghttp2/issues/2194

I'm not sure that I have the correct combination of `fails_with`.

Missing runtime dependency in latest linux CI run
```
 /home/linuxbrew/.linuxbrew/Cellar/nghttp2/1.62.1/bin/nghttp: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by /home/linuxbrew/.linuxbrew/Cellar/nghttp2/1.62.1/bin/nghttp)
```

File: Formula/n/nghttp2.rb
Comment (cho-m) [reply Formula/n/nghttp2.rb:43]: > > That's less of an issue and generally should work OK even though `libnghttp2` use an older GCC.
> 
> I'd have thought we'd be ok using a newer GCC only for `nghttp2` and not `libnghttp2`, but the `test` block disagrees, unfortunately.

Can try adding `Formula["gcc@13"].opt_lib/"gcc/13"` to `RPATH`. Not sure if it will precede `#{HOMEBREW_PREFIX}/lib/gcc/current` (otherwise may have some extra missing linkage reports whenever brew `gcc` is installed on CI).

---

EDIT: Though we may want to wait for upstream response if they will support GCC 11 given it is commonly used in LTS distros.

File: Formula/n/nghttp2.rb
Comment (MikeMcQuaid) [inline Formula/n/nghttp2.rb:38]: What macOs version does this correspond to?

File: Formula/n/nghttp2.rb
Comment (MikeMcQuaid) [reply Formula/n/nghttp2.rb:43]: > Adding a dependency with `libnghttp2` would have massive effects on the dependency tree of homebrew-core as a whole so wouldn't really be acceptable without discussion.

Is this just on Linux?

> If `libnghttp2` is broken on older Clang, we must drop support for macOS 10.11 in Homebrew/brew.

Why is that, can you elaborate?

Comment (Bo98) [reply Formula/n/nghttp2.rb:43]: > I'd have thought we'd be ok using a newer GCC only for `nghttp2` and not `libnghttp2`, but the `test` block disagrees, unfortunately.

`libnghttp2` doesn't use C++ at all.

> Why is that, can you elaborate?

Because it's a dependency for curl, and we require brewed curl on older macOS.

Comment (carlocab) [reply Formula/n/nghttp2.rb:43]: Actually, compiling `nghttp2` (and not `libnghttp2`) with Homebrew `gcc` should just work. It'll fail the Linux-only GCC dependency audit, though.

Comment (MikeMcQuaid) [reply Formula/n/nghttp2.rb:43]: > Because it's a dependency for curl, and we require brewed curl on older macOS.

Gotcha, thanks. Yeh, I think it'd either be: drop it as a dependency, drop it as a dependency on old macOS versions (dangerous with no CI coverage) or drop those macOS versions.

> It'll fail the Linux-only GCC dependency audit, though.

Seems like a good case to opt-out of.

Comment (carlocab) [reply Formula/n/nghttp2.rb]: Let's see if using `gcc` does indeed work before we add an audit opt-out... dbded26c93988ede0fd5be08d4430babba95f34e

Comment (carlocab) [reply Formula/n/nghttp2.rb]: Ok, that works. Homebrew/brew#17384

File: CODEOWNERS
Comment (carlocab) [inline CODEOWNERS:19]:
```suggestion
audit_exceptions/linux_only_gcc_dependency_allowlist.json               @Homebrew/tsc @Bo98
```
@Bo98 you ok if we do this instead?

Comment (MikeMcQuaid) [reply CODEOWNERS:19]:
```suggestion
audit_exceptions/linux_only_gcc_dependency_allowlist.json               @Homebrew/tsc
```

I think it's fine for it to be just the TSC here.

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good once we figure decode on the `CODEOWNERS` (or move to another PR).

**Pull request closed without merging**
